src/Plot/PlotController.purs

```diff
@@ -0,0 +1,54 @@
+module Plot.PlotController where
+
+import Prelude
+import Draw.Commands (DrawCommand)
+import Draw.Actions (drawPlotLine)
+import Plot.Commands (PlotCommand(..))
+import Data.Traversable (for_)
+import Data.Array ((..), zipWith, tail)
+import Data.Either (Either(..))
+import Data.Int (floor, toNumber)
+import Data.Maybe (fromMaybe)
+import Effect (Effect)
+import Effect.Aff (Aff, Canceler, Error, makeAff, nonCanceler)
+import Types (Size, XYBounds, Position)
+import Plot.GridLines (clearAndDrawGridLines)
+
+computePlotAsync :: Size -> PlotCommand -> Aff (DrawCommand Unit)
+computePlotAsync canvasSize plot = makeAff $ runComputation canvasSize plot
+
+runComputation :: Size -> PlotCommand -> (Either Error (DrawCommand Unit) -> Effect Unit) -> Effect Canceler
+runComputation canvasSize (Plot shouldClear bounds func) callback = do
+  callback $ Right
+    $ do
+        -- Computation for drawing plot here
+        if shouldClear then
+          clearAndDrawGridLines bounds
+        else
+          pure unit
+        plotSimpleLine canvasSize bounds func
```

Comment:
Note for future:  Originally I was thinking that we would compute asynchronously only the enclosure plot and compute the commands for approximate plot and things like the grid lines synchronously.  Nevertheless, it may turn out that it is best to continue computing them asynchronously.

Reply:
That is definitely something that we should look into